pallets/asset-manager/src/tests.rs

```diff
@@ -547,3 +551,126 @@ fn set_min_xcm_fee_should_work() {
         );
     })
 }
+
+fn create_asset_and_location(
+    token: &str,
+) -> Option<(AssetRegistryMetadata<Balance>, AssetLocation)> {
+    let manta_asset_metadata = create_asset_metadata(token, token, 12, 1u128, false, false);
+    let manta_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::new(
+        1,
+        X2(
+            Parachain(2015),
+            GeneralKey(WeakBoundedVec::force_from(token.as_bytes().to_vec(), None)),
+        ),
+    )));
+    Some((manta_asset_metadata, manta_location))
+}
+
+#[test]
+fn register_lp_asset_should_work() {
+    new_test_ext().execute_with(|| {
+        // Register first non native token.
+        let (manta_asset_metadata8, manta_location8) = create_asset_and_location("Asset8").unwrap();
+        assert_ok!(AssetManager::register_asset(
+            RuntimeOrigin::root(),
+            manta_location8.clone(),
+            manta_asset_metadata8.clone()
+        ));
+        assert_eq!(
+            AssetIdLocation::<Runtime>::get(8),
+            Some(manta_location8.clone())
+        );
+
+        assert_noop!(
+            AssetManager::register_lp_asset(
+                RuntimeOrigin::root(),
+                8,
+                8,
+                manta_location8.clone(),
+                manta_asset_metadata8.clone()
+            ),
+            Error::<Runtime>::AssetIdNotDifferent
+        );
+        assert_noop!(
+            AssetManager::register_lp_asset(
+                RuntimeOrigin::root(),
+                8,
+                9,
+                manta_location8,
+                manta_asset_metadata8
+            ),
+            Error::<Runtime>::AssetIdNotExist
+        );
+
+        let (manta_asset_metadata9, manta_location9) = create_asset_and_location("Asset9").unwrap();
```

Comment:
i think you can wrap this logic in a function and dedupe - register_and_assert

Reply:
one of testcase assert register failed because of `AssetAlreadyRegistered`. if wrap the login into one function, we still need to deal with failed case. I think it's fine now.